app/src/main/java/com/github/sdp_begreen/begreen/activities/MainActivity.kt

```diff
@@ -69,70 +75,54 @@ class MainActivity : AppCompatActivity() {
+     * @param descriptionTW The view that contains the description in the drawer
      */
-    private fun setupDrawerUserInfo() {
-        if (drawerInitialized) return
-
-        drawerInitialized = true
-        val imageView: ImageView = findViewById(R.id.nav_drawer_profile_picture_imageview)
+    private fun setupDrawerUserInfo(
+        imageView: ImageView, usernameTW: TextView, descriptionTW: TextView
+    ) {
         if (connectedUserViewModel.currentUser.value == null) {
-            setUpUserNameAndDescription(null) // call with null to set default values
+            setUpUserNameAndDescription(null, usernameTW, descriptionTW) // call with null to set default values
         }
         if (connectedUserViewModel.currentUserProfilePicture.value == null) {
             imageView.setImageBitmap(
                 BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture))
         }
 
-        connectedUserViewModel.currentUser.observe(this) {
-            setUpUserNameAndDescription(it)
-        }
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                launch {
+                    connectedUserViewModel.currentUser.collect {
+                        setUpUserNameAndDescription(it, usernameTW, descriptionTW)
+                    }
+                }
 
-        connectedUserViewModel.currentUserProfilePicture.observe(this) {
-            imageView.setImageBitmap(
-                it ?:
-                BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture)
-            )
+                launch {
+                    connectedUserViewModel.currentUserProfilePicture.collect {
+                        imageView.setImageBitmap(it ?:
+                        BitmapFactory.decodeResource(resources, R.drawable.blank_profile_picture))
+                    }
+                }
+            }
         }
     }
 
-    /**
-     * Helper method to get the currently logged in user if it exists
-     */
-    private suspend fun getConnectedUser(): User? =
-        Firebase.auth.currentUser?.uid?.let { FirebaseDB.getUser(it) }
-
     /**
      * Helper method to set the username and the description of a user if it exists
+     *
+     * @param user The user from whom to display information
+     * @param usernameTW The view that contains the username in the drawer
+     * @param descriptionTW The view that contains the description in the drawer
      */
-    private fun setUpUserNameAndDescription(user: User?) {
-        findViewById<TextView>(R.id.nav_drawer_username_textview).text =
-            user?.displayName ?: getString(R.string.nav_drawer_username)
-
-        findViewById<TextView>(R.id.nav_drawer_description_textview).text =
-            user?.description ?: getString(R.string.nav_drawer_user_description)
+    private fun setUpUserNameAndDescription(
+        user: User?, usernameTW: TextView, descriptionTW: TextView
```

Comment:
TW for... TextWiew?

Reply:
Aïe that's a big mistakes from me. I inverted `v` and `w`. I will change that thanks.